tools/clang/test/CodeGenSPIRV/intrinsics.all.hlsl

```diff
@@ -83,5 +85,40 @@ void main() {
     // CHECK-NEXT: OpStore %result [[all_float4]]
     float4 l;
     result = all(l);
-}
 
+    // CHECK-NEXT: [[m:%\d+]] = OpLoad %float %m
+    // CHECK-NEXT: [[mat1x1_to_bool:%\d+]] = OpFOrdNotEqual %bool [[m]] %float_0
+    // CHECK-NEXT: OpStore %result [[mat1x1_to_bool]]
+    float1x1 m;
+    result = all(m);
+
+    // CHECK-NEXT: [[n:%\d+]] = OpLoad %v3float %n
+    // CHECK-NEXT: [[mat1x3_to_bool:%\d+]] = OpFOrdNotEqual %bool [[n]] [[v3float_0]]
+    // CHECK-NEXT: [[all_mat1x3:%\d+]] = OpAll %bool [[mat1x3_to_bool]]
+    // CHECK-NEXT: OpStore %result [[all_mat1x3]]
+    float1x3 n;
+    result = all(n);
+
+    // CHECK-NEXT: [[o:%\d+]] = OpLoad %v2float %o
+    // CHECK-NEXT: [[mat2x1_to_bool:%\d+]] = OpFOrdNotEqual %bool [[o]] [[v2float_0]]
```

Comment:
The result type for this comparison seems incorrect. It should be `%v2bool` I think.